### PR TITLE
Bugfix: Summary component—bail out of initialization of component when content is too brief; refresh max-height on resize; hide overlay at desktop

### PR DESCRIPTION
**Pull request by anselmbradford**
## Changes

- Bail out of Summary component initialization when the amount of content is less than the default summary content height.
- Re-calculate max-height when resizing window.
- Hide gradient overlay when it is not in use.

## How to test this PR

1. Pull branch and  `gulp clean && gulp build`
2. Visit http://localhost:8000/ask-cfpb/how-can-i-tell-if-a-friend-neighbor-or-family-member-is-a-victim-of-financial-exploitation-en-1933/ and see that there is a Summary component on mobile.
2. Visit http://localhost:8000/ask-cfpb/can-my-bankcredit-union-charge-me-a-fee-to-use-another-bank-or-credit-unions-atm-en-1081/ and see there is no Summary component on mobile.
3. Visit http://localhost:8000/ask-cfpb/what-is-a-credit-reporting-company-en-1251/ and see that you can click the "Go to our browsable list..." links. Resize page to narrow mobile size and see that content isn't cut off when summary is expanded.

### CODE REVIEW

**Review by csebianlander — APPROVED** [review]

**Comment by Scotchester** [comment]
> 3\. Visit http://localhost:8000/ask-cfpb/can-my-bankcredit-union-charge-me-a-fee-to-use-another-bank-or-credit-unions-atm-en-1081/ and see there is no Summary component on mobile.

It's not a huge deal, certainly not a blocker for this PR if it's hard to overcome, but when I load this page without cache, I notice a very brief flash of the Summary component and then it disappears.

File: cfgov/unprocessed/js/organisms/Summary.js [hunk 1]
```diff
@@ -39,6 +39,13 @@ function Summary( element ) {
       return this;
     }
 
+    /* Bail out of initializatiion if the height of the summary's content
+       is less then our summary height of 5.5ems (16 * 5.5 = 88) */
```
Comment (Scotchester) [inline cfgov/unprocessed/js/organisms/Summary.js:43]: Maybe include a URL reference to where this is defined in Less?

**Review by Scotchester — APPROVED** [review]
Working great. One tiny suggestion.

File: cfgov/unprocessed/js/organisms/Summary.js (hunk 1 above)
Comment (anselmbradford) [reply cfgov/unprocessed/js/organisms/Summary.js]: Added in https://github.com/cfpb/consumerfinance.gov/pull/6252/commits/9424b2ee90c6b75dfb1ded28ac2c343f724a8a7d

Also added `u-hidden` on the buttons by default to hide the flash on load you mentioned above.

**Pull request merged**